src/gasnow_websocket.rs

```diff
@@ -0,0 +1,172 @@
+use crate::{
+    gasnow::{self, ResponseData},
+    GasPriceEstimating,
+};
+use anyhow::{bail, ensure, Result};
+use futures::StreamExt;
+use serde_json::Value;
+use std::time::{Duration, Instant};
+use tokio::sync::watch;
```

Comment:
I was confused about the receiver code because in my head it would either:
- block because no one was receiving the sent messages (if it were a buffered channel)
- grow forever for the same reason (if if were an unbounded)

Then, I learned about `watch` channels. Very cool stuff!

Reply:
Yes, watch is nice. Initially I had Arc<Mutex<...>> until I remembered watch. Watch is nice because it understands updates so you can wait on the next update for example. Also it allows the task to know when all receivers have been dropped.